File: src/app/lib/models/buff_category.dart
/// Categorises the type of buff/bonus effect.
enum BuffCategory {
  multiplier('multiplier'),
  duration('duration'),
  spawn('spawn'),
  probability('probability'),
  trade('trade'),
  weather('weather'),
  other('other');

  const BuffCategory(this.value);
  final String value;

  static BuffCategory fromJson(String json) =>
      BuffCategory.values.firstWhere(
        (e) => e.value == json,
        orElse: () => BuffCategory.other,
      );

  String toJson() => value;
}

File: src/backend/GoCalGo.Contracts/Events/BuffCategory.cs
using System.Text.Json.Serialization;

namespace GoCalGo.Contracts.Events
{
    /// <summary>
    /// Categorises the type of buff/bonus effect.
    /// Serialised as lowercase kebab-case strings.
    /// </summary>
    [JsonConverter(typeof(JsonStringEnumConverter<BuffCategory>))]
    public enum BuffCategory
    {
        [JsonStringEnumMemberName("multiplier")]
        Multiplier,

        [JsonStringEnumMemberName("duration")]
        Duration,

        [JsonStringEnumMemberName("spawn")]
        Spawn,

        [JsonStringEnumMemberName("probability")]
        Probability,

        [JsonStringEnumMemberName("trade")]
        Trade,

        [JsonStringEnumMemberName("weather")]
        Weather,

        [JsonStringEnumMemberName("other")]
        Other,
    }
}
